File: ios/objc/StorylyDemo/StorylyDemo/ViewController.m
//
//  ViewController.m
//  StorylyDemo
//
//  Created by Levent ORAL on 18.02.2020.
//  Copyright © 2020 App Samurai Inc. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet StorylyView *storylyView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.storylyView.storylyInit = [[StorylyInit alloc] initWithStorylyId:@YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD];
    self.storylyView.rootViewController = self;
    self.storylyView.delegate = self;
}

-(void)storylyLoaded:(StorylyView *)storylyView storyGroupList:(NSArray<StoryGroup *> *)storyGroupList {
    NSLog(@"storylyLoaded");
}

- (void)storylyLoadFailed:(StorylyView *)storylyView errorMessage:(NSString *)errorMessage {
    NSLog(@"storylyLoadFailed");
}

- (BOOL)storylyActionClicked:(StorylyView *)storylyView rootViewController:(UIViewController *)rootViewController story:(Story *)story {
    NSLog(@"storylyActionClicked");
    return YES;
}

- (void)storylyStoryPresented:(StorylyView *)storylyView {
    NSLog(@"storylyStoryPresented");
}

- (void)storylyStoryDismissed:(StorylyView *)storylyView {
    NSLog(@"storylyStoryDismissed");
}

- (void)storylyUserInteracted:(StorylyView *)storylyView storyGroup:(StoryGroup *)storyGroup story:(Story *)story storyComponent:(StoryComponent *)storyComponent {
    switch (storyComponent.type) {
        case StoryComponentTypeQuiz:
            {
                StoryQuizComponent *quizComponent = (StoryQuizComponent *)storyComponent;
                // quizComponent actions
            }
            break;
        case StoryComponentTypePoll:
            {
                StoryPollComponent *pollComponent = (StoryPollComponent *)storyComponent;
                // pollComponent actions
            }
            break;
        case StoryComponentTypeEmoji:
            {
                StoryEmojiComponent *emojiComponent = (StoryEmojiComponent *)storyComponent;
                // emojiComponent actions
            }
            break;
        case StoryComponentTypeUndefined:
            break;
        default:
            break;
    }
}

@end

File: ios/swift/StorylyDemo/StorylyDemo/ViewController.swift
//
//  ViewController.swift
//  StorylyDemo
//
//  Created by Levent ORAL on 25.09.2019.
//  Copyright © 2019 App Samurai Inc. All rights reserved.
//

import UIKit
import Storyly

class ViewController: UIViewController {

    @IBOutlet weak var storylyView: StorylyView!
    private var customLoadingView = CustomLoadingView(frame: CGRect(x: 0, y: 0, width: 100, height: 100))
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.storylyView.storylyInit = StorylyInit(storylyId: @YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD)
        self.storylyView.rootViewController = self
        self.storylyView.delegate = self
        // If you want your own loading view uncomment and edit the following lines
//        self.storylyView.storylyLoadingView = customLoadingView
    }
}

extension ViewController: StorylyDelegate {
    func storylyLoaded(_ storylyView: StorylyView, storyGroupList: [StoryGroup]) {
        print("storylyLoaded")
    }
    
    func storylyLoadFailed(_ storylyView: StorylyView, errorMessage: String) {
        print("storylyLoadFailed")
    }
    
    func storylyActionClicked(_ storylyView: Storyly.StorylyView, rootViewController: UIViewController, story: Storyly.Story) -> Bool {
        print("storylyActionClicked")
        
        // Edit and use the following method to open an external custom view
//        self.showCustomExternalView(storylyView: storylyView, story: story)
        
        return true // return false if sdk should handle click
    }
    
    func storylyStoryPresented(_ storylyView: StorylyView) {
        print("storylyStoryPresented")
    }
    
    func storylyStoryDismissed(_ storylyView: StorylyView) {
        print("storylyStoryDismissed")
    }
    
    func storylyUserInteracted(_ storylyView: StorylyView, storyGroup: StoryGroup, story: Story, storyComponent: StoryComponent) {
        switch storyComponent.type {
            case .Quiz:
                if let quizComponent = storyComponent as? StoryQuizComponent {
                    // quizComponent actions
                }
            case .Poll:
                if let pollComponent = storyComponent as? StoryPollComponent {
                    // pollComponent actions
                }
            case .Emoji:
                if let emojiComponent = storyComponent as? StoryEmojiComponent {
                    // emojiComponent actions
                }
            case .Undefined: do {}
            default: do {}
        }
    }
}
